ddpui/ddpairbyte/airbytehelpers.py

```diff
@@ -500,14 +513,96 @@ def get_one_connection(org: Org, connection_id: str):
 
 
 def reset_connection(org: Org, connection_id: str):
-    """reset an airbyte connection"""
-    if not OrgTask.objects.filter(
+    """
+    reset the connection via the sync deployment
+    this will run a full reset + sync
+    """
+    org_server_block = OrgPrefectBlockv1.objects.filter(org=org, block_type=AIRBYTESERVER).first()
+
+    if not org_server_block:
+        logger.error("Airbyte server block not found")
+        return None, "airbyte server block not found"
+
+    sync_org_task = OrgTask.objects.filter(
         org=org,
         connection_id=connection_id,
-    ).exists():
+        task__slug=TASK_AIRBYTESYNC,
+    ).first()
+    if not sync_org_task:
         return None, "connection not found"
 
-    airbyte_service.reset_connection(connection_id)
+    sync_dataflow_orgtask = DataflowOrgTask.objects.filter(
+        orgtask=sync_org_task, dataflow__dataflow_type="manual"
+    ).first()
+
+    if sync_dataflow_orgtask is None:
+        logger.error("Sync dataflow not found")
+        return None, "sync dataflow not found"
+
+    reset_org_task = OrgTask.objects.filter(
+        org=org, connection_id=connection_id, task__slug=TASK_AIRBYTERESET
+    ).first()
+
+    if not reset_org_task:
+        logger.error("Reset OrgTask not found")
+        return None, "reset OrgTask not found"
+
+    # full reset + sync; run via the manual sync deployment
+    params = {
+        "config": {
+            "tasks": [
+                setup_airbyte_sync_task_config(reset_org_task, org_server_block, seq=1).to_json(),
+                setup_airbyte_sync_task_config(sync_org_task, org_server_block, seq=2).to_json(),
+            ],
+            "org_slug": org.slug,
+        }
+    }
+
+    # check if the connection is "large" for scheduling
+    connection_meta = ConnectionMeta.objects.filter(connection_id=connection_id).first()
+    is_connection_large_enough = (
+        True == connection_meta.schedule_large_jobs if connection_meta else False
+    )
+
+    schedule_at = None
+    job: ConnectionJob = None
+    if is_connection_large_enough:
+        schedule_at = get_schedule_time_for_large_jobs()
+        # if there is a flow run scheduled , delete it
+        job = ConnectionJob.objects.filter(
```

Comment:
1. the lines `job = ... if job: ...` can be pulled out of `if is_connection_large_enough`

thinking of a case where we update `is-large-enough` in the meta when there is already a job scheduled

3. after `delete_flow_run` can we update `job.flow_run_id = None` and save

thinking of a case when scheduling the next one fails after deletion succeeds

Reply:
1. we dont want to delete a job if the connections is not large enough. So how would taking it out help ? Correct if i misunderstood your comment

3. done